astropy/coordinates/angles/core.py

```diff
@@ -403,36 +404,16 @@ def _wrap_at(self, wrap_angle):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
-        a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = wrap_angle.to_value(self.unit)
         self_angle = self.view(np.ndarray)
-        if NUMPY_LT_2_0:
-            # Ensure ndim>=1 so that comparison is done using the angle dtype.
-            self_angle = self_angle[np.newaxis]
+        a360 = np.asarray(u.degree.to(self.unit, 360.0), dtype=self_angle.dtype)
+        wrap_angle = wrap_angle.to_value(self.unit).astype(
+            self_angle.dtype, copy=COPY_IF_NEEDED
+        )
+        if not self_angle.data.readonly:
```

Comment:
I would prefer to write this logic in a flatter style, such as
```python
if self_angle.data.readonly and needs_wrapping(self_angle, wrap_angle, a360).any():
    raise ValueError(...)

wrap_at(...)
```

Reply:
Nice suggestion, will change